components/ml/org.wso2.carbon.ml.database/src/main/java/org/wso2/carbon/ml/database/internal/MLDatabaseService.java

```diff
@@ -2637,4 +2637,35 @@ public void updateSamplePoints(long datasetVersionId, SamplePoints samplePoints)
             MLDatabaseUtils.closeDatabaseResources(connection, updateStatement);
         }
     }
+    
+    @Override
+    public boolean isValidModelStatus(long modelId, int tenantId, String userName) throws DatabaseHandlerException {
+        Connection connection = null;
+        ResultSet result = null;
+        PreparedStatement statement = null;
+        try {
+            connection = dbh.getDataSource().getConnection();
+            statement = connection.prepareStatement(SQLQueries.GET_MODEL_STATUS);
+            statement.setLong(1, modelId);
+            statement.setInt(2, tenantId);
+            statement.setString(3, userName);
+            result = statement.executeQuery();
+            if (result.first()) {
+                if (result.getString(1).equalsIgnoreCase(MLConstants.MODEL_STATUS_COMPLETE)) {
```

Comment:
As a good practice, always compare values retrieved from other software components/systems with known constants to avoid redundant null checks.